doc/userguide/capture-hardware/ebpf-xdp.rst

```diff
@@ -0,0 +1,318 @@
+eBPF and XDP
+============
+
+Introduction
+------------
+
+eBPF stands for extended BPF. This is an extended version of Berkeley Packet Filter available in recent
+Linux kernel versions.
+
+It provides more advanced features with eBPF programs developed in C and able to use structured data shared
+between kernel and userspace.
+
+eBPF is used for three things in Suricata:
+
+- eBPF filter: any BPF like filter can be developed. An example of filter accepting only packet for some VLANs is provided.
+- eBPF load balancing: provide programmable load balancing. A simple ippair load balancing is provided.
+- XDP programs: suricata can load XDP programs. A bypass program is provided.
+
+Bypass can be implemented in eBPF and XDP. The advantage of XDP is that the packets are dropped at the earliest stage
+possible. So performance are better. But bypassed packets don't reach the network so you can't use this on regular
+traffic but only on duplicated/sniffed traffic.
+
+XDP
+~~~
+
+XDP provides another Linux native way of optimising Suricata's performance on sniffing high speed networks.
+
+::
+
+ XDP or eXpress Data Path provides a high performance, programmable network data path in the Linux kernel as part of the IO Visor Project. XDP provides bare metal packet processing at the lowest point in the software stack which makes it ideal for speed without compromising programmability. Furthermore, new functions can be implemented dynamically with the integrated fast path without kernel modification.
+
+`More info about XDP <https://www.iovisor.org/technology/xdp>`__
+
+Requirements
+------------
+
+- Newer kernel that supports XDP (4.13.10 in our case)
+- RSS symmetric hashing on the NIC ( Intel 82599ES 10-Gigabit/x520/x540 in our case)
+- In tree kernel drivers NIC drivers
```

Comment:
I'm not 100% sure if everyone knows what you mean with that, can you rephrase it?